apps/rush-lib/src/utilities/JsonFile.ts

```diff
@@ -7,29 +7,72 @@ import * as jju from 'jju';
 import Utilities from './Utilities';
 
 /**
+ * Options for JsonFile.saveJsonFile()
+ *
+ * @public
+ */
+export interface ISaveJsonFileOptions {
+  /**
+   * If there is an existing file, and the contents have not changed, then
+   * don't write anything; this preserves the old timestamp.
+   */
+  onlyIfChanged?: boolean;
+}
+
+/**
  * Utilities for reading/writing JSON files.
  * @public
  */
 export default class JsonFile {
-
-  /* tslint:disable:no-any */ // JSON objects are dynamically typed
+  /**
+   * Loads a JSON file.
+   */
+  // tslint:disable-next-line:no-any
   public static loadJsonFile(jsonFilename: string): any {
     if (!fsx.existsSync(jsonFilename)) {
       throw new Error(`Input file not found: ${jsonFilename}`);
     }
 
     const buffer: Buffer = fsx.readFileSync(jsonFilename);
     try {
-      return jju.parse(buffer.toString());
+      const jsonData: Object = jju.parse(buffer.toString());
+      return jsonData;
     } catch (error) {
       throw new Error(`Error reading "${jsonFilename}":` + os.EOL + `  ${error.message}`);
     }
   }
 
-  public static saveJsonFile(jsonData: any, jsonFilename: string): void {
+  /**
+   * Saves the file to disk.  Returns false if nothing was written due to options.onlyIfChanged.
+   */
+  // tslint:disable-next-line:no-any
+  public static saveJsonFile(jsonData: any, jsonFilename: string, options?: ISaveJsonFileOptions): boolean {
```

Comment:
If I do that, then the caller could still do saveJsonFile(x,y,undefined)

---
In reply to: [116346856](https://github.com/Microsoft/web-build-tools/pull/187#discussion_r116346856) [](ancestors = 116346856)